extension/test/integration/get-payment-methods.handler.spec.js

```diff
@@ -0,0 +1,211 @@
+const { expect } = require('chai')
+const _ = require('lodash')
+
+const ctpClientBuilder = require('../../src/ctp/ctp-client')
+const iTSetUp = require('./integration-test-set-up')
+const c = require('../../src/config/constants')
+
+describe('::getPaymentMethods::', () => {
+  let ctpClient
+
+  before(async () => {
+    ctpClient = ctpClientBuilder.get()
+    await iTSetUp.initServerAndExtension(ctpClient)
+  })
+
+  after(async () => {
+    await iTSetUp.cleanupResources(ctpClient)
+  })
+
+  it('given a payment ' +
+    'when getPaymentMethodsRequest custom field is set and custom field getPaymentMethodsResponse is not set ' +
+    'then should set custom field getPaymentMethodsResponse ' +
+    'and interface interaction with type getPaymentMethods', async () => {
+    const getPaymentMethodsRequestDraft = {
+      countryCode: "DE",
+      shopperLocale: "de-DE",
+      amount: {
+        currency: "EUR",
+        value: 1000
+      }
+    }
+    const paymentDraft = {
+      amountPlanned: {
+        currencyCode: 'EUR',
+        centAmount: 1000,
+      },
+      paymentMethodInfo: {
+        paymentInterface: c.CTP_ADYEN_INTEGRATION
+      },
+      custom: {
+        type: {
+          typeId: "type",
+          key: c.CTP_WEB_COMPONENTS_PAYMENT_TYPE_KEY
+        },
+        fields: {
+          getPaymentMethodsRequest: JSON.stringify(getPaymentMethodsRequestDraft)
+        }
+      }
+    }
+
+    const { statusCode, body: payment } = await ctpClient.create(ctpClient.builder.payments, paymentDraft)
+    expect(statusCode).to.equal(201)
+
+    const { getPaymentMethodsRequest, getPaymentMethodsResponse } = payment.custom.fields
+    expect(getPaymentMethodsRequest).to.be.deep.equal(JSON.stringify(getPaymentMethodsRequestDraft))
+
+    const interfaceInteraction = payment.interfaceInteractions
+      .find(interaction => interaction.fields.type === c.CTP_INTERACTION_TYPE_GET_PAYMENT_METHODS)
+    expect(interfaceInteraction).to.not.undefined
+    expect(getPaymentMethodsResponse).to.be.deep.equal(interfaceInteraction.fields.response)
+
+    expect(JSON.parse(interfaceInteraction.fields.request)).to.be.deep.equal({
+      merchantAccount: process.env.ADYEN_MERCHANT_ACCOUNT,
+      ...getPaymentMethodsRequestDraft
+    })
+
+    const interfaceInteractionResponse = JSON.parse(interfaceInteraction.fields.response)
+    expect(interfaceInteractionResponse.groups).to.be.an.instanceof(Array)
+    expect(interfaceInteractionResponse.paymentMethods).to.be.an.instanceof(Array)
+    expect(interfaceInteractionResponse.additionalData).to.not.exist
+  })
+
+  it('given an existing payment' +
```

Comment:
I don't think we need these error cases as integration tests. It's enough to have them as unit test where you mock call to adyen. If we keep adding integration tests, it will take very long time to run in the end.

Reply:
This project is not handling too many cases on integration tests, so performance is not a problem here yet, and IMO it will not be a problem in the future. also, those tests are not slow. But I could consider adding those as a unit test as you said those parts really depend on the Adyen and unit test could be a better candidate, yes.

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/232/commits/a48be471fd6b07f2f29f90297e6cc16a7038a94b

I am skipping the last test, as I could not decide yet where to put json invalid validations, for the things which I did not fix in this PR, added as a todo for not to forget later.